plugins/out_stackdriver/stackdriver_http_request.c

```diff
@@ -0,0 +1,453 @@
+                          flb_sds_len(http_request->protocol));
+
+    /* Integer sub-fields */
+    msgpack_pack_str(mp_pck, 11);
+    msgpack_pack_str_body(mp_pck, "requestSize", 11);
+    msgpack_pack_int64(mp_pck, http_request->requestSize);
+
+    msgpack_pack_str(mp_pck, 12);
+    msgpack_pack_str_body(mp_pck, "responseSize", 12);
+    msgpack_pack_int64(mp_pck, http_request->responseSize);
+
+    msgpack_pack_str(mp_pck, 6);
+    msgpack_pack_str_body(mp_pck, "status", 6);
+    msgpack_pack_int64(mp_pck, http_request->status);
+
+    msgpack_pack_str(mp_pck, 14);
+    msgpack_pack_str_body(mp_pck, "cacheFillBytes", 14);
+    msgpack_pack_int64(mp_pck, http_request->cacheFillBytes);
+
+    /* Boolean sub-fields */
+    msgpack_pack_str(mp_pck, 11);
+    msgpack_pack_str_body(mp_pck, "cacheLookup", 11);
+    if (http_request->cacheLookup == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+
+    msgpack_pack_str(mp_pck, 8);
+    msgpack_pack_str_body(mp_pck, "cacheHit", 8);
+    if (http_request->cacheLookup == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+
+    msgpack_pack_str(mp_pck, 30);
+    msgpack_pack_str_body(mp_pck, "cacheValidatedWithOriginServer", 30);
+    if (http_request->cacheValidatedWithOriginServer == FLB_TRUE) {
+        msgpack_pack_true(mp_pck);
+    }
+    else {
+        msgpack_pack_false(mp_pck);
+    }
+}
+
+/* latency should be in the format: 
+ *      whitespace (opt.) + integer + point & decimal (opt.)
+ *      + whitespace (opt.) + "s" + whitespace (opt.) 
+ */
+static void validate_latency(msgpack_object_str latency_in_payload, 
+                             struct http_request_field *http_request) {
+    flb_sds_t pattern = flb_sds_create("^\\s*\\d\\d*(.\\d\\d*)?\\s*s\\s*$");
+    char extract_latency[latency_in_payload.size];
+    struct flb_regex *regex;
+
+    int status = 0;
+    int i = 0, j = 0;
+
+    regex = flb_regex_create(pattern);
+    status = flb_regex_match(regex, latency_in_payload.ptr, latency_in_payload.size);
+    flb_regex_destroy(regex);
+    flb_sds_destroy(pattern);
+
+    if (status == 1) {
+        for (; i < latency_in_payload.size; ++ i) {
+            if (latency_in_payload.ptr[i] == '.' || latency_in_payload.ptr[i] == 's' 
+                || isdigit(latency_in_payload.ptr[i])) {
+                extract_latency[j] = latency_in_payload.ptr[i];
+                ++ j;
+            }
+        }
+        http_request->latency = flb_sds_copy(http_request->latency, extract_latency, j);
+    }
+}
+                                                                                        
+/* Return true if httpRequest extracted */
+int extract_http_request(struct http_request_field *http_request, 
+                         msgpack_object *obj, int *extra_subfields)
+{
+    http_request_status op_status = NO_HTTPREQUEST;
```

Comment:
Could this just be a boolean?

Reply:
Yes. I just want to get sync with other features like insertId. WDYT?